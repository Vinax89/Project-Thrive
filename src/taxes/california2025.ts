export function estimateCaliforniaTax(income: number) {
  const brackets = [
    { upTo: 10412, rate: 0.01 },
    { upTo: 24684, rate: 0.02 },
    { upTo: 38959, rate: 0.04 },
    { upTo: 54082, rate: 0.06 },
    { upTo: 68350, rate: 0.08 },
    { upTo: 349137, rate: 0.093 },
    { upTo: 418961, rate: 0.103 },
    { upTo: 698271, rate: 0.113 },
    { upTo: Infinity, rate: 0.123 }
  ];
  let tax = 0, prev = 0;
  for (const b of brackets) {
    const amt = Math.max(0, Math.min(income, b.upTo) - prev);
    tax += amt * b.rate;
    prev = b.upTo;
    if (income <= b.upTo) break;
  }
  return Math.round(tax * 100) / 100;
}
