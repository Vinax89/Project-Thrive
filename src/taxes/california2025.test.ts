import { describe, it, expect } from 'vitest';
import { estimateCaliforniaTax } from './california2025';

describe('estimateCaliforniaTax', () => {
  const boundaries = [10412, 24684, 38959, 54082, 68350, 349137, 418961, 698271];

  it('calculates tax at bracket boundaries', () => {
    const taxes = boundaries.map((income) => estimateCaliforniaTax(income));
    expect(taxes).toMatchSnapshot();
  });

  it('returns 0 for no income', () => {
    expect(estimateCaliforniaTax(0)).toBe(0);
  });

  it('calculates tax in the top bracket', () => {
    expect(estimateCaliforniaTax(1_000_000)).toBe(104989.14);
  });
});
