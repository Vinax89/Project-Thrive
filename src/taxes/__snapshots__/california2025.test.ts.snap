// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`estimateCaliforniaTax > calculates tax at bracket boundaries 1`] = `
[
  104.12,
  389.56,
  960.56,
  1867.94,
  3009.38,
  29122.57,
  36314.44,
  67876.47,
]
`;
